lib/amazon-pricing/helpers/instance-type.rb

```diff
@@ -225,13 +243,20 @@ def nf_to_size
           "10xlarge" => 80,
           "12xlarge" => 96,
           "16xlarge" => 128,
-          "metal"    => 128, # temporary (for _direct_ users of this hash), as only applies to i3.metal
+          "metal" => 128,  # We will be removing this once size_to_nf is deprecated.
           "18xlarge" => 144,
           "24xlarge" => 192,
           "32xlarge" => 256,
       }
       NF_TO_SIZE_TABLE = SIZE_TO_NF_TABLE.invert
 
+      METAL_TO_NF_TABLE = {
+        'u-6tb1.metal' => 896,
+        'u-9tb1.metal' => 896,
+        'u-12tb1.metal' => 896,
+        'i3.metal' => 128,
```

Comment:
why adding this entry here if we try to get the largest size in a family?

Reply:
why run the code, if we know the NF for i3.metal is 128? For the others, those NF values aren't stored in the size_to_nf table.